apps/addons/tests/test_models.py

```diff
@@ -2033,12 +2028,45 @@ def manifest(self, basename):
                             'addons', basename)
 
     def test_blacklisted_guid(self):
+        """New deletions won't be added to BlacklistedGuid but legacy support
+        should still be tested."""
         BlacklistedGuid.objects.create(guid='guid@xpi')
         with self.assertRaises(forms.ValidationError) as e:
             Addon.from_upload(self.get_upload('extension.xpi'),
                               [self.platform])
         eq_(e.exception.messages, ['Duplicate add-on ID found.'])
 
+    def test_existing_guid(self):
+        # Create deleted addon with same guid
+        deleted = Addon.from_upload(self.get_upload('extension.xpi'),
+                                    [self.platform])
+        deleted.update(status=amo.STATUS_PUBLIC)
+        deleted.delete()
+        eq_(deleted.guid, 'guid@xpi')
```

Comment:
We've switched to pytest, so we use simple `assert`s now:
`assert deleted.guid == 'guid@xpi'`